libraries/protocol/steem_operations.cpp

```diff
@@ -76,13 +76,52 @@ namespace steemit { namespace protocol {
       }
    }
 
+   struct comment_options_extension_validate_visitor
+   {
+      comment_options_extension_validate_visitor() {}
+
+      typedef void result_type;
+
+      void operator()( const comment_payout_beneficiaries& cpb ) const
+      {
+         cpb.validate();
+      }
+   };
+
+   void comment_payout_beneficiaries::validate()const
+   {
+      uint32_t sum = 0;
+
+      FC_ASSERT( beneficiaries.size(), "Must specify at least one beneficiary" );
+      FC_ASSERT( beneficiaries.size() < 128, "Cannot specify more than 127 beneficiaries." ); // Require size serializtion fits in one byte.
+
+      string_less str_cmp;
+
+      for( size_t i = 0; i < beneficiaries.size() - 1; i++ )
```

Comment:
This loop looks dangerous because it depends upon line 95, consider changing it to be 

``` 
for( size_t i = 1; i < beneficiars.size(); ++i )
```

And then update internals to use i-1 and i rather than i and i+1